src/api/contract/localtunneldebugger/v1.ts

```diff
@@ -0,0 +1,15 @@
+// This module is contractual and should not be changed after release.
+// It should be in sync with vscode-kubernetes-tools-api/ts/localtunneldebugger/v1.ts
+// at all times.
+
+export interface LocalTunnelDebuggerV1 {
+    registerLocalTunnelDebugProvider(localTunnelDebugProvider: LocalTunnelDebuggerV1.LocalTunnelDebugProvider): void;
+    startLocalTunnelDebugProvider(target?: any): void;
```

Comment:
Is the idea of this to allow an extension to trigger local tunnel debugging?  If so, maybe drop `Provider` from the name?

Reply:
Oh I see, e.g. "startLocalTunnelDebugging"? Yes, the idea is this actually starts the debug provider's debugging method.

Reply:
I went with "startLocalTunnelDebugSession", but I'm open to changes ofc.

Reply:
If this starts the provider's session shouldn't it be a method on the provider rather than a method exposed by the Kubernetes extension?

I feel I don't have this straight in my mind yet.  Could you explain _who_ calls this method (the k8s extension, the provider, the extension containing the provider, or a third-party extension) and _when_?  Thanks and apologies for not getting it.

Reply:
No worries at all, it could definitely be a naming issue or that I'm not using the correct pattern. Basically, in extension.ts we bind the Debug (Local Tunnel) command to the method exposed by the k8s extension: 

```
registerCommand('extension.vsKubernetesDebugLocalTunnel', (target?: any) => { localTunnelDebugger.startLocalTunnelDebugSession(target); }),
```
And then inside the implementation of the method in component/localtunneldebugger.ts, we see if a provider is installed, do some tie-breaking if multiple providers are available, and then call into the provider that we chose:

```
    startLocalTunnelDebugSession(target?: any) {
        const providerSetting: string | undefined = getLocalTunnelDebugProvider();
        const providerName: string | undefined = providerSetting ? providerSetting : this.providers.map((p) => p.id).sort()[0];
        const providerToUse: LocalTunnelDebugProvider | undefined = this.providers.find((p) => p.id === providerName);

        // On success start early.
        if (providerToUse) {
            providerToUse.startLocalTunnelDebugging(target);
            return;
        }
```
Let me know if that makes sense, or if I can flatten this in a better way!

Reply:
It makes sense, but that makes it sound like this is a method that is called only from _within_ the extension (from `extension.ts`), not by external consumers of the API.  In which case it should not be part of the API, just a plain exported function from the implementation module.  Unless the _provider_ extension needs to call it, there's no need to put it in that part of the API.

(More generally, unless a method gets called across extension boundaries - core calling into provider or provider calling into core - then it doesn't need to be in the API at all.  But this interface is specifically one that the provider uses to call into the core.)

Hope this makes sense - very happy to tee up a call if that would be a more efficient way to get us on the same page.

Reply:
Hey Ivan, that does make sense! Thanks for clarifying that. I don't know why I was confusing what needed to be in the API vs. what simply needs to be exported. :) I've removed the method from the interface and I tested to make sure everything was still able to snap together like it should, but don't hesitate to let me know if there's anything else I can fix, stylistically or otherwise.